File: testBack/src/main/java/com/bookadmin/modules/book/entity/Book.java
package com.bookadmin.modules.book.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;

/**
 * 图书,与前端 src/types/book.ts 的 Book 对应。
 * categoryName 为关联展示字段,不入库。
 */
@Data
@TableName("book")
public class Book {

    @TableId(type = IdType.AUTO)
    private Long id;
    private String title;
    private String author;
    private String isbn;
    private Long categoryId;
    private String publisher;
    private BigDecimal price;
    private Integer stock;
    private String cover;
    private String status;
    private String description;
    private LocalDateTime createdAt;

    @TableField(exist = false)
    private String categoryName;
}

File: test/src/types/book.ts
// 图书相关类型
import type { PageParams } from './api';

export type BookStatus = 'on' | 'off';

/** 图书 */
export interface Book {
  id: number;
  title: string;
  author: string;
  isbn: string;
  categoryId: number;
  categoryName: string;
  publisher: string;
  price: number;
  stock: number;
  cover: string;
  status: BookStatus;
  description: string;
  createdAt: string;
}

/** 图书查询参数 */
export interface BookQuery extends PageParams {
  keyword?: string;
  categoryId?: number;
  status?: BookStatus;
}

/** 图书新增/编辑表单 */
export type BookForm = Omit<Book, 'id' | 'categoryName' | 'createdAt'>;
